### DISCUSSION TITLE
Project state regarding past release of Camunda 8

### DISCUSSION THREAD

**Discussion by mickroll** [wkspower/wks-platform#151] [General]
I am fairly new to camunda, and wks-platform looks like a promising implementation base for new projects. Great work!

As Camunda 8 was released mid 2022, i am wondering if a migration to it is planned for the near future.

I must admit, a migration does look complicated, as a lot has changed. https://docs.camunda.io/docs/8.0/guides/migrating-from-camunda-platform-7/
As i understand, a migration would need a rewrite of some core components (the camunda API changed, for example the concept of `ExternalTaskHandler` was replaced by `ZeebeWorker`). Additionally, the underlying process engine was replaced by zeebe (i don't know what was used before, though). Process-Models need to be migrated. All in all, this seems like a lot of work, especially if not only code but also existing data of process instances needs to be migrated.

So, of course, before I decide to use wks-platform in its current state, I'm trying to assess the pros and cons of waiting until Camunda 8 is integrated.

@francav What puzzles me a bit is that Camnuda 8 was released about 3 months before the first commit to wks-platform. Might i ask, why you stuck to Camunda 7 back then? Was C8 not ready or lacked features you needed at the time? I am curious to understand the whole Camunda ecosystem a bit better.

**Comment by francav** [comment]
Hi @mickroll I am sorry for the delay. Outlook was sending github messages to my "other" inbox =/
Will answer your questions in a few.

**Reply by francav** [reply]
@Hafflgav

**Comment by francav** [comment]
Thanks for your words @mickroll I am glad you are considering WKS Platform as a base for your projects.
This project was designed with compatibility for multiple process engines in mind. See the following java modules for reference:

https://github.com/wkspower/wks-platform/tree/main/apps/java/libraries/bpm-engine-interface (bpm concepts abstraction)
https://github.com/wkspower/wks-platform/tree/main/apps/java/libraries/bpm-engine-client (bpm engine client abstraction)
https://github.com/wkspower/wks-platform/tree/main/apps/java/libraries/bpm-engine-c7-client (Camunda 7 client implementation - **Ready**)
https://github.com/wkspower/wks-platform/tree/main/apps/java/libraries/bpm-engine-c8-client (Camunda 8 client implementation - **WIP**)

**Comment by francav** [comment]
Hence, our goal is to have it integrated with C7, C8 and other bpm engines

**Comment by mickroll** [comment]
Ah, that sheds some light. Thank you very much for the explanation. Do you have a time horizon for the implementation of the camunda 8 client?

**Reply by francav** [reply]
Would you like to help on that?

**Reply by francav** [reply]
I can help on explaining the requirements

**Reply by francav** [reply]
Should we connect? www.linkedin.com/in/victorfranca
@mickroll

**Reply by francav** [reply]
Camunda 8 integration completed @mickroll
